resolver/type_syntax.cc

```diff
@@ -247,31 +247,52 @@ ParsedSig parseSigWithSelfTypeParams(core::MutableContext ctx, ast::Send *sigSen
                     sig.seen.params = true;
 
                     if (send->args.empty()) {
-                        break;
-                    }
-
-                    if (send->args.size() > 1) {
                         if (auto e = ctx.beginError(send->loc, core::errors::Resolver::InvalidMethodSignature)) {
-                            e.setHeader("Wrong number of args to `{}`. Expected: `{}`, got: `{}`", send->fun.show(ctx),
-                                        "0-1", send->args.size());
+                            auto paramsStr = send->fun.show(ctx);
+                            e.setHeader("`{}` must be given arguments", paramsStr);
+
+                            core::Loc loc{ctx.file, send->loc};
+                            auto orig = loc.source(ctx);
+                            auto dot = orig.rfind(".");
+                            if (orig.rfind(".") == string::npos) {
+                                e.replaceWith("Remove this use of `params`", loc, "");
```

Comment:
This case would produce fewer errors if it also searched after the send for a `.`. Right now it will suggest replacing `params.void` with just `.void`.